schema.graphql

```diff
@@ -107,6 +121,162 @@ type UnprocessedEntity @entity {
+  unstakedAtBlock: Block
+  stakeMsgs: [AppStakeMsg] @derivedFrom(field: "application")
+  unstakeMsgs: [AppUnstakeMsg] @derivedFrom(field: "application")
+  delegateMsgs: [DelegateToGatewayMsg] @derivedFrom(field: "application")
+  undelegateMsgs: [UndelegateToGatewayMsg] @derivedFrom(field: "application")
+  sourceTransferMsgs: [TransferApplicationMsg] @derivedFrom(field: "sourceApplication")
+  destinationTransferMsgs: [TransferApplicationMsg] @derivedFrom(field: "destinationApplication")
+  sourceTransferBeginEvents: [TransferApplicationBeginEvent] @derivedFrom(field: "source")
+  destinationTransferBeginEvents: [TransferApplicationBeginEvent] @derivedFrom(field: "destination")
+  sourceTransferEndEvents: [TransferApplicationEndEvent] @derivedFrom(field: "source")
+  destinationTransferEndEvents: [TransferApplicationEndEvent] @derivedFrom(field: "destination")
+  sourceTransferErrorEvents: [TransferApplicationErrorEvent] @derivedFrom(field: "source")
+  destinationTransferErrorEvents: [TransferApplicationErrorEvent] @derivedFrom(field: "destination")
+  unstakingBeginEvents: [ApplicationUnbondingBeginEvent] @derivedFrom(field: "application")
+  unstakingEndEvents: [ApplicationUnbondingEndEvent] @derivedFrom(field: "application")
+  transferredFromAt: Block
+  transferFrom: Application
+  transferTo: Application
+  transferToEndedAt: Block
+  # Account because we don't know if the destination is an application
+  transferringTo: Account
+  transferEndsAtHeight: BigInt
+}
+
+type ApplicationDelegatedToGateway @entity {
+  id: ID!
+  gateway: Gateway!
+  application: Application!
+}
+
+type ApplicationService @entity {
+  id: ID!
+  application: Application!
+  service: Service!
+}
+
+type TransferApplicationBeginEvent @entity {
+  # The ID of the event result of getEventId(event)
+  id: ID!
+  source: Application!
+  destination: Application!
+  transaction: Transaction!
+  block: Block!
+}
+
+type TransferApplicationEndEvent @entity {
+  # The ID of the event result of getEventId(event)
+  id: ID!
+  source: Application!
+  destination: Application!
+  transaction: Transaction
+  block: Block!
+}
+
+type TransferApplicationErrorEvent @entity {
+  # The ID of the event result of getEventId(event)
+  id: ID!
+  source: Application!
+  destination: Account!
+  error: String!
+  transaction: Transaction
+  block: Block!
+}
+
+type ApplicationUnbondingBeginEvent @entity {
+  # The ID of the event result of getEventId(event)
+  id: ID!
+  application: Application!
+  # 0 = ELECTIVE, 1 = BELOW_MIN_STAKE, 2 = TRANSFERRED
+  reason: Int!
+  sessionEndHeight: BigInt!
+  unstakingEndHeight: BigInt!
+  transaction: Transaction
+  block: Block!
+}
+
+type ApplicationUnbondingEndEvent @entity {
+  # The ID of the event result of getEventId(event)
+  id: ID!
+  application: Application!
+  # 0 = ELECTIVE, 1 = BELOW_MIN_STAKE, 2 = TRANSFERRED
+  reason: Int!
```

Comment:
(same as other enums; let's prefer gql enums and resort to typescript enums only if necessary and #PUC why if so)

This particular enum is already a protobuf enum, I'm not sure if that's useful in the context but please do consider it. I think we should prefer that above all else (as it's an existing and canonical type) if it is feasible.

Reply:
yes